### PR TITLE
Switch to newer kind - fix bazel build issue

### PR DESCRIPTION
**Pull request by dims**
Trying to fix failure in CI job:
https://testgrid.k8s.io/conformance-all#conformance-image,%20master%20%28dev%29

Pulling in fix in kind:
https://github.com/kubernetes-sigs/kind/pull/963

### CODE REVIEW

File: experiment/kind-conformance-image-e2e.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ set -o xtrace
 # - get some fixes related to Kubernetes's build changing
 # - don't get surprised when kind changes between now and the next stable release
 # We should switch back to a release tag at the next release.
-STABLE_KIND_VERSION=v0.5.1
+STABLE_KIND_VERSION=v0.6.0
```
Comment (BenTheElder) [inline experiment/kind-conformance-image-e2e.sh]: we should switch kubeconfig management like https://github.com/kubernetes/test-infra/pull/15247

File: experiment/kind-conformance-image-e2e.sh [hunk 2]
```diff
@@ -128,7 +128,7 @@ run_tests() {
 
   VERSION=$(echo -n "${KUBE_GIT_VERSION}" | cut -f 1 -d '+')
   export VERSION
-  KUBECONFIG=$(kind get kubeconfig-path)
+  KUBECONFIG="${HOME}/.kube/kind-config-default"
   export KUBECONFIG
```
Comment (BenTheElder) [inline experiment/kind-conformance-image-e2e.sh]: This should be before create cluster

Comment (dims) [reply experiment/kind-conformance-image-e2e.sh]: Done

File: experiment/kind-conformance-image-e2e.sh (hunk 1 above)
Comment (dims) [reply experiment/kind-conformance-image-e2e.sh]: Done

**Comment by dims** [comment]
is this better @BenTheElder ?

**Comment by BenTheElder** [comment]
/lgtm
/approve
In other scripts I'd do kubeconfig early in main but I'm not sure we even need it, it seems like only CI runs this?

**Pull request merged**